docs/source/spec-workload-executor.rst

```diff
@@ -37,55 +37,133 @@ Behavioral Description
+   to communicate to the workload executor, and ultimately the unified test
+   runner, that they should stop running operations.
+
+#. MUST invoke the unified test runner to execute the workload.
+   If the workload includes a ``loop`` operation, the workload will run until
+   terminated by the workload executor; otherwise, the workload will terminate
+   when the unified test runner finishes executing all of the operations.
+   The workload executor MUST handle the case of a non-looping workload and
+   it MUST terminate if the unified test runner completely executes the
+   specified workload.
+   
+   If the unified test runner raises an error while executing the workload,
+   the error MUST be reported using the same format as errors handled by the
+   unified test runner, as described in the unified test runner specification
+   under the ``loop`` operation. Errors handled by the workload
+   executor MUST be included in the calculated (and reported) error count.
+   
+   If the unified test runner reports a failure while executing the workload,
+   the failure MUST be reported using the same format as failures handled by the
+   unified test runner, as described in the unified test runner specification
+   under the ``loop`` operation. Failures handled by the workload
+   executor MUST be included in the calculated (and reported) failure count.
+   If the driver's unified test runner is intended to handle all failures
+   internally, failures that propagate out of the unified test runner MAY
+   be treated as errors by the workload executor.
+
+#. Upon receipt of the termination signal, MUST instruct the
+   unified test runner to stop looping, as defined in the unified test format.
+
+#. MUST wait for the unified test runner to finish executing.
+   
+#. MUST use the unified test runner to retrieve the following
+   entities by name from the entity map, if they are set:
+   
+   * ``iterations``: the number of iterations that the workload executor
+     performed over the looped operations. If the iteration count was not
+     reported by the test runner, such as because the respective option was
+     not specified in the test scenario, the workload executor MUST use
+     ``-1`` as the number of iterations.
+   
+   * ``successes``: the number of successful operations that the workload
+     executor performed over the looped operations. If the iteration count
+     was not reported by the test runner, such as because the respective
+     option was not specified in the test scenario, the workload executor
+     MUST use ``-1`` as the number of successes.
+   
+   * ``errors``: array of documents describing the errors that occurred
+     while the workload executor was executing the operations.
+   
+   * ``failures``: array of documents describing the failures that occurred
+     while the workload executor was executing the operations.
+   
+   * ``events``: array of documents describing the CMAP events that occurred
```

Comment:
IIRC, this is no longer just CMAP events. Does this also warrant changes to the section below?

> MUST write the collected events, errors and failures into a JSON file named ``events.json`` in the current directory...

Reply:
Yes, fixed.